src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ticketRuleForm.tsx

```diff
@@ -19,18 +19,23 @@ import {FormField} from 'app/views/settings/projectAlerts/issueEditor/ruleNode';
 type Props = {
   formFields: {[key: string]: any};
   instance?: IssueAlertRuleAction | IssueAlertRuleCondition;
-  onSubmitAction: (data: {[key: string]: string}) => void;
+  onSubmitAction: (
+    data: {[key: string]: string},
+    dynamicFieldChoices: {[key: string]: string[]}
+  ) => void;
   onPropertyChange: (rowIndex: number, name: string, value: string) => void;
   index: number;
 };
 
 type State = {
   showModal: boolean;
+  dynamicFieldChoices: any;
```

Comment:
@mgaeta we can probably be stricter with the types. We can create a new type:
``` type dynamicFieldChoices = {[key: string]: string[]}``` and use this type instead.